draft-normington-jsonpath-latest.txt

```diff
@@ -464,7 +487,52 @@ Semantics
 
    A negative index "j" selects an element of an array of length "len"
    if and only if "0 <= j + len < len", in which case it selects the
-   same element as selected by the non-negative index "j + len".
+   same element as the non-negative index "j + len".
+
+   An array slice matches any node which is an array and selects any
+   elements of the array indexed by the members of the slice, as defined
+   below, in the same order as the slice.  It does not match a node
+   which is not an array.
```

Comment:
Should "does not match a node which is not an array" appear at the top of the section with the introduction about Array Access?

Reply:
Yes. I was trying to avoid duplication in the syntax and semantics section, but I think it reads better with a bit of repetition.

When my wife was a maths teacher, she used to tell me "You only learn by repetition. You only learn by repetition."